test/e2e/serial/tests/tolerations.go

```diff
@@ -433,6 +438,181 @@ var _ = Describe("[serial][disruptive][slow][rtetols] numaresources RTE tolerati
+					_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+					Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+
+					By("verify there is a running RTE pod on the tainted node")
+					pods, err = podlist.With(fxt.Client).ByDaemonset(ctx, *ds)
+					Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Daemonset %q: %v", ds.Name, err)
+					Expect(len(pods)).To(Equal(len(workers)), "pods number is not as expected for daemonset %s/%s: expected %d found %d", ds.Namespace, ds.Name, len(workers)-1, len(pods))
+					found := false
+					for _, pod := range pods {
+						if pod.Spec.NodeName == taintedNode.Name {
+							found = true
+							break
+						}
+					}
+					Expect(found).To(BeTrue())
+				})
+
+				It("[test_id:72855][reboot_required][slow][tier2] should tolerate node taint on NROP CR creation", func(ctx context.Context) {
+					fxt.IsRebootTest = true
+					By("add tolerations to NROP CR to tolerate the taint - no RTE running yet on any node")
+					nropNewObj := nroOperObj.DeepCopy()
+					nropNewObj.ObjectMeta = metav1.ObjectMeta{
+						Name: nroOperObj.Name,
+					}
+					if nropNewObj.Spec.NodeGroups[0].Config == nil {
+						nropNewObj.Spec.NodeGroups[0].Config = &nropv1.NodeGroupConfig{}
+					}
+					nropNewObj.Spec.NodeGroups[0].Config.Tolerations = testToleration()
+
+					err := fxt.Client.Create(ctx, nropNewObj)
+					Expect(err).ToNot(HaveOccurred())
+
+					mcps, err := nropmcp.GetListByNodeGroupsV1(ctx, e2eclient.Client, nropNewObj.Spec.NodeGroups)
+					Expect(err).ToNot(HaveOccurred())
+					waitForMcpUpdate(fxt.Client, ctx, mcps)
+
+					klog.Info("waiting for DaemonSet to be ready")
+					ds, err := wait.With(fxt.Client).Interval(time.Second).Timeout(time.Minute).ForDaemonsetPodsCreation(ctx, dsKey, len(workers))
+					Expect(err).NotTo(HaveOccurred(), "pods number is not as expected for daemonset: expected %d found %d", len(workers), ds.Status.DesiredNumberScheduled)
+					_, err = wait.With(e2eclient.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+					Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+
+					By("verify RTE pods are running on all worker nodes including the tainted node")
+					pods, err := podlist.With(fxt.Client).ByDaemonset(ctx, *ds)
+					Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Daemonset %q: %v", ds.Name, err)
+					Expect(len(pods)).To(Equal(len(workers)), "pods number is not as expected for daemonset %s/%s: expected %d found %d", ds.Namespace, ds.Name, len(workers)-1, len(pods))
+					found := false
+					for _, pod := range pods {
+						if pod.Spec.NodeName == taintedNode.Name {
+							found = true
+							break
+						}
+					}
+					Expect(found).To(BeTrue())
```

Comment:
again this seems to be a nice candidate for the helper function to find a RTE pod among all pods running on a node